### DISCUSSION TITLE
iMet vs RS41

### DISCUSSION THREAD

**Discussion by gurlavie** [dl9rdz/rdz_ttgo_sonde#147] [Q&A]
Just a beginner question, why is the iMet decoding not feasible ? compared to RS41?

sorry if this is too trivial, I guess the modulation is the problem, but searched and couldn't find an answer.

**Comment by dl9rdz** [comment]
All supported radiosondes use direct FSK modulation, simply speaking the sonde transmissions alternates between two different frequencies directly (such as 401.005 MHz and 400.995 MHz) to transmit "0" and "1" bits. The SX1278 chip on the TTGO is a FSK decoder that provides the demodulated output (0 or 1) to the software.

The iMet uses AFSK modulation, i.e., to transmit a 0 or 1 bit, it creates an audio signal such as 1200 Hz and 2200 Hz, which then is FM modulated. To decode that, you should have an FM demodulator and then process the audio signal, i.e. detect 1200 Hz or 2200 Hz signals.  But the FM demodulated audio signal is not available on the SX1278.

**Comment by bazjo** [comment]
I don't know whether this has been brought up in this repo yet, but only the iMet radiosondes from the american branch (iMet-4) seem to use AFSK modulation. The radiosondes from the south african branch (iMet-5x) seem to use GFSK modulation with parameters similar to RS41 (https://intermet.co/radiosondes/imet-54/)

**Comment by dl9rdz** [comment]
Interesting. In that case of course imet-5x could be supported.
